src/main/java/Majima/Parser.java

```diff
@@ -0,0 +1,106 @@
+     */
+    public static Command parse(String input) throws MajimaException {
+        String[] parts = input.split(" ", 2);
+        String command = parts[0];
+        String args = (parts.length > 1) ? parts[1] : "";
+
+        switch (command) {
+            case "list":
+            return new ListCommand();
+            case "todo":
+            return parseAddCommand(Task.TaskType.TODO, args);
+            case "deadline":
+            return parseAddCommand(Task.TaskType.DEADLINE, args);
+            case "event":
+            return parseAddCommand(Task.TaskType.EVENT, args);
+            case "delete":
+            return parseDeleteCommand(args);
+            case "mark":
+            return parseMarkCommand(args);
+            case "unmark":
+            return parseUnmarkCommand(args);
+            case "bye":
+            return new ExitCommand();
+            case "find":
+            return new FindCommand(args);
+            default:
+            throw new MajimaException("Kiryu? I ain't got the faintest idea of what ya just said!");
+        }
+    }
+
+    private static Command parseAddCommand(Task.TaskType type, String args) throws MajimaException {
+        String description;
+        switch (type) {
+            case TODO:
+                if (args.trim().isEmpty()) {
+                    throw new MajimaException("Kiryu! Your description can't be empty!");
+                }
+                return new AddTodoCommand(args);
+            case DEADLINE:
+                String[] parts = args.split(" /by ", 2); // for Deadline
+                if (parts.length < 2) {
+                    throw new MajimaException("Kiryu! The deadline command must include a deadline date! Use /by!");
+                }
+                description = parts[0].trim();
+                String deadline = parts[1].trim();
+                return new AddDeadlineCommand(description, deadline);
+            case EVENT:
+                parts = args.split(" /from | /to ", 3);
+                if (parts.length < 3) {
+                    throw new MajimaException("Kiryu! The event command must include a start and end time, using /from and /to!");
+                }
+                description = parts[0].trim();
+                String from = parts[1].trim();
+                String to = parts[2].trim();
+                return new AddEventCommand(description, from, to);
+            default:
+                throw new MajimaException("Unknown task type found: " + type);
+        }
+    }
+
+    private static Command parseDeleteCommand(String args) throws MajimaException {
+        try {
+            int index = Integer.parseInt(args) - 1;
+            return new DeleteCommand(index);
+        } catch (NumberFormatException e) {
+            throw new MajimaException("Eh? Kiryu-chan, that ain't no number!");
+        }
+    }
+
+    private static int parseTaskNumber (String args) {
```

Comment:
might want to remove the space between the function and parameter

Reply:
Fixed and pushed, thanks for catching this for me.